.github/workflows/build-test.yml

```diff
@@ -269,6 +269,121 @@ jobs:
+    runs-on: ubuntu-latest
+    needs: [build, check-files, check-is-hotfix, load-e2e-cctp-files]
+    if: needs.check-files.outputs.run_tests == 'true' && needs.check-is-hotfix.outputs.is_hotfix == 'false'
+    strategy:
+      fail-fast: false # If one test fails, let the other tests run
+      matrix:
+        tests:
+          ${{ fromJson(needs.load-e2e-cctp-files.outputs.matrix) }}
+
+    steps:
+      - name: Free Disk Space (Ubuntu)
+        uses: jlumbroso/free-disk-space@main
+        with:
+          # this might remove tools that are actually needed,
+          # if set to "true" but frees about 6 GB
+          tool-cache: false
+          
+          # all of these default to true, but feel free to set to
+          # "false" if necessary for your workflow
+          android: true
+          dotnet: true
+          haskell: true
+          large-packages: true
+          docker-images: false
+          swap-storage: true
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install node_modules
+        uses: OffchainLabs/actions/node-modules/install@main
+
+      - name: Restore build artifacts
+        uses: ./.github/actions/build-artifacts/restore
+
+      - name: Install cypress
+        run: yarn cypress install
+
+      - name: Install linux dependencies
+        run: |
+          sudo apt-get install --no-install-recommends -y \
+          fluxbox \
+          xvfb
+
+      - name: Run xvfb and fluxbox
+        run: |
+          Xvfb :0 -screen 0 1366x768x24 -listen tcp -ac &
+          fluxbox &
+        env:
+          DISPLAY: :0.0
+
+      - name: Run e2e tests via cypress-io/github-action
+        uses: cypress-io/github-action@8d3918616d8ac34caa2b49afc8b408b6a872a6f5 # pin@v6.7.1
+        with:
+          start: yarn start
+          command: 'yarn test:e2e:cctp --browser chromium'
+          wait-on: http://127.0.0.1:3000
+          wait-on-timeout: 120
+          spec: ./packages/arb-token-bridge-ui/tests/e2e/specs/*
+        env:
+          DISPLAY: :0.0
+          TEST_FILE: ${{ matrix.tests.file }}
+          SKIP_METAMASK_SETUP: true
+          CYPRESS_RECORD_VIDEO: ${{ matrix.tests.recordVideo }}
+          PRIVATE_KEY_CCTP: ${{ secrets.E2E_PRIVATE_KEY_CCTP }}
+          PRIVATE_KEY_USER: ${{ secrets.E2E_PRIVATE_KEY_USER }}
+          NEXT_PUBLIC_IS_E2E_TEST: true
+          NEXT_PUBLIC_INFURA_KEY: ${{ secrets.NEXT_PUBLIC_INFURA_KEY }}
+
+      - name: Archive e2e artifacts
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: e2e-artifacts-pull-request-${{ github.event.pull_request.number }}-commit-${{ github.sha }}-cctp
+          path: |
+            ./packages/arb-token-bridge-ui/cypress/videos
+            ./packages/arb-token-bridge-ui/cypress/screenshots
+        continue-on-error: true
+
+      - name: Throw error if tests failed
+        if: steps.e2e-cctp-run.outputs.status == 'failed'
+        run: exit 1
+
+  test-e2e-cctp-success:
```

Comment:
While this works, I feel there is quite some repetition in the setup and steps for normal e2e vs cctp e2e, and I think it's worth exploring if we can reuse the step logic across both the tests.